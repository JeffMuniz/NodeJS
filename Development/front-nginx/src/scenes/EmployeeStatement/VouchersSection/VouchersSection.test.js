import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import VoucherSection from "./VouchersSection";

jest.mock("@common", () => ({
  SvgIcon: "SvgIcon",
}));

const defaultProps = {
  vouchers: [
    {
      id: "1",
      idProduct: "1",
      type: "food",
      printedName: "teste",
      issueDate: "01/01/2019",
      cardNumber: "1111 22xx xxxx 4444",
      activationDate: "Wed Sep 12 2018 15:00:04 GMT-0300",
    },
    {
      printedName: "teste",
      id: "1",
      idProduct: "1",
      type: "food",
      issueDate: "01/01/2019",
      cardNumber: "1111 22xx xxxx 4444",
      activationDate: "Wed Sep 12 2018 15:00:04 GMT-0300",
    },
    {
      printedName: "teste",
      id: "2",
      idProduct: "2",
      type: "meal",
      issueDate: "01/01/2019",
      cardNumber: "1111 22xx xxxx 4444",
      activationDate: "Wed Sep 12 2018 15:00:04 GMT-0300",
    },
  ],
  updateVouchers: jest.fn(),
  changeVoucherStatus: jest.fn(),
};

describe("Voucher Section - Unit Test", () => {
  it("Should render Voucher Section type food", () => {
    // given
    const renderer = new ShallowRenderer();

    // when
    renderer.render(<VoucherSection {...defaultProps} />);
    const result = renderer.getRenderOutput();

    // then
    expect(result).toMatchSnapshot();
  });
});
