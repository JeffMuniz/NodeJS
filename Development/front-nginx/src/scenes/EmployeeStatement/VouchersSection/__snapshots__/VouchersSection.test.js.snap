// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Voucher Section - Unit Test Should render Voucher Section type food 1`] = `
<UNDEFINED>
  <Styled(styled.div)>
    <VoucherListComponent
      active={0}
      changeVoucherStatus={[MockFunction]}
      items={
        Array [
          Object {
            "activationDate": "Wed Sep 12 2018 15:00:04 GMT-0300",
            "cardNumber": "1111 22xx xxxx 4444",
            "id": "1",
            "idProduct": "1",
            "issueDate": "01/01/2019",
            "printedName": "teste",
            "type": "food",
          },
          Object {
            "activationDate": "Wed Sep 12 2018 15:00:04 GMT-0300",
            "cardNumber": "1111 22xx xxxx 4444",
            "id": "1",
            "idProduct": "1",
            "issueDate": "01/01/2019",
            "printedName": "teste",
            "type": "food",
          },
          Object {
            "activationDate": "Wed Sep 12 2018 15:00:04 GMT-0300",
            "cardNumber": "1111 22xx xxxx 4444",
            "id": "2",
            "idProduct": "2",
            "issueDate": "01/01/2019",
            "printedName": "teste",
            "type": "meal",
          },
        ]
      }
      updateVouchers={[MockFunction]}
    />
  </Styled(styled.div)>
</UNDEFINED>
`;
